[2020-09-01 18:02:31,634] INFO Expiring session 0x10009b1b5100001, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2020-09-01 18:02:30,641] WARN Attempting to send response via channel for which there is no open connection, connection id 192.168.56.1:9092-192.168.56.1:63195-0 (kafka.network.Processor)
[2020-09-01 18:02:30,644] INFO Unable to read additional data from server sessionid 0x10009b1b5100001, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:33,810] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:33,921] INFO Socket connection established, initiating session, client: /127.0.0.1:49321, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,009] INFO Invalid session 0x10009b1b5100001 for client /127.0.0.1:49321, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2020-09-01 18:02:34,011] WARN Unable to reconnect to ZooKeeper service, session 0x10009b1b5100001 has expired (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,017] INFO EventThread shut down for session: 0x10009b1b5100001 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,034] INFO Unable to reconnect to ZooKeeper service, session 0x10009b1b5100001 has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,048] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2020-09-01 18:02:34,057] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2020-09-01 18:02:34,223] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@35229f85 (org.apache.zookeeper.ZooKeeper)
[2020-09-01 18:02:34,247] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2020-09-01 18:02:34,254] INFO zookeeper.request.timeout value is 0. feature enabled= (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,319] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2020-09-01 18:02:34,426] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,470] INFO Socket connection established, initiating session, client: /127.0.0.1:49425, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,509] INFO Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x10009b1b5100002, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:02:34,550] INFO Stat of the created znode at /brokers/ids/0 is: 771,771,1598963554547,1598963554547,1,0,0,72068252889513986,200,0,771
 (kafka.zk.KafkaZkClient)
[2020-09-01 18:02:34,600] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://DESKTOP-JP0D9RM:9092, czxid (broker epoch): 771 (kafka.zk.KafkaZkClient)
[2020-09-01 18:10:39,911] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:10:39,911] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:10:39,912] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:10:39,913] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:13:18,746] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:23:18,746] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 18:29:23,851] INFO Unable to read additional data from server sessionid 0x10009b1b5100002, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:24,844] INFO Expiring session 0x10009b1b5100002, timeout of 18000ms exceeded (org.apache.zookeeper.server.ZooKeeperServer)
[2020-09-01 18:29:25,505] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,517] INFO Socket connection established, initiating session, client: /127.0.0.1:51918, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,528] INFO Invalid session 0x10009b1b5100002 for client /127.0.0.1:51918, probably expired (org.apache.zookeeper.server.ZooKeeperServer)
[2020-09-01 18:29:25,528] WARN Unable to reconnect to ZooKeeper service, session 0x10009b1b5100002 has expired (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,530] INFO [ZooKeeperClient Kafka server] Session expired. (kafka.zookeeper.ZooKeeperClient)
[2020-09-01 18:29:25,530] INFO EventThread shut down for session: 0x10009b1b5100002 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,548] INFO [ZooKeeperClient Kafka server] Initializing a new session to localhost:2181. (kafka.zookeeper.ZooKeeperClient)
[2020-09-01 18:29:25,541] INFO Unable to reconnect to ZooKeeper service, session 0x10009b1b5100002 has expired, closing socket connection (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,552] INFO Initiating client connection, connectString=localhost:2181 sessionTimeout=18000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@35229f85 (org.apache.zookeeper.ZooKeeper)
[2020-09-01 18:29:25,559] INFO jute.maxbuffer value is 4194304 Bytes (org.apache.zookeeper.ClientCnxnSocket)
[2020-09-01 18:29:25,559] INFO zookeeper.request.timeout value is 0. feature enabled= (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,561] INFO Creating /brokers/ids/0 (is it secure? false) (kafka.zk.KafkaZkClient)
[2020-09-01 18:29:25,563] INFO Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,565] INFO Socket connection established, initiating session, client: /127.0.0.1:51921, server: localhost/127.0.0.1:2181 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,570] INFO Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x10009b1b5100003, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 18:29:25,575] INFO Stat of the created znode at /brokers/ids/0 is: 776,776,1598965165573,1598965165573,1,0,0,72068252889513987,200,0,776
 (kafka.zk.KafkaZkClient)
[2020-09-01 18:29:25,576] INFO Registered broker 0 at path /brokers/ids/0 with addresses: PLAINTEXT://DESKTOP-JP0D9RM:9092, czxid (broker epoch): 776 (kafka.zk.KafkaZkClient)
[2020-09-01 19:39:33,286] INFO [GroupCoordinator 0]: Member consumer-1-83bdda45-cfef-4a10-99f4-d442eee35fe8 in group KafkaExampleConsumer has failed, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,308] INFO [GroupCoordinator 0]: Preparing to rebalance group KafkaExampleConsumer in state PreparingRebalance with old generation 162 (__consumer_offsets-10) (reason: removing member consumer-1-83bdda45-cfef-4a10-99f4-d442eee35fe8 on heartbeat expiration) (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,318] INFO [GroupCoordinator 0]: Group KafkaExampleConsumer with generation 163 is now empty (__consumer_offsets-10) (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,399] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group KafkaExampleConsumer in Empty state. Created a new member id consumer-1-d17e1ba4-8a96-4e4e-b4ad-d54440b8ff2d for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,454] INFO Unable to read additional data from server sessionid 0x10009b1b5100003, likely server has closed socket, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
[2020-09-01 19:39:33,456] INFO [GroupCoordinator 0]: Preparing to rebalance group KafkaExampleConsumer in state PreparingRebalance with old generation 163 (__consumer_offsets-10) (reason: Adding new member consumer-1-d17e1ba4-8a96-4e4e-b4ad-d54440b8ff2d with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,667] INFO [GroupCoordinator 0]: Stabilized group KafkaExampleConsumer generation 164 (__consumer_offsets-10) (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:33,926] INFO [GroupCoordinator 0]: Assignment received from leader for group KafkaExampleConsumer for generation 164 (kafka.coordinator.group.GroupCoordinator)
[2020-09-01 19:39:34,631] INFO Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error) (org.apache.zookeeper.ClientCnxn)
[2020-09-01 19:39:34,644] INFO Socket connection established, initiating session, client: /0:0:0:0:0:0:0:1:50447, server: localhost/0:0:0:0:0:0:0:1:2181 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 19:39:34,670] INFO Session establishment complete on server localhost/0:0:0:0:0:0:0:1:2181, sessionid = 0x10009b1b5100003, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2020-09-01 19:42:07,065] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,065] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,065] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,066] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,066] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,066] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:42:07,066] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:43:18,716] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 0 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2020-09-01 19:43:33,854] WARN Session 0x10009b1b5100003 for server localhost/0:0:0:0:0:0:0:1:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at java.base/sun.nio.ch.SocketDispatcher.read0(Native Method)
	at java.base/sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at java.base/sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:276)
	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:233)
	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:223)
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:358)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:75)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:363)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1223)
